### PR TITLE
escape literal percent sign when formatting

### LINKED ISSUE #61503: `kubectl explain` doesn't parse % well
All `%` in the field description becomes `%!(MISSING)`:

```console
$ kubectl explain deploy.spec.strategy.rollingUpdate
[...]
FIELDS:
   maxSurge     <string>
     The maximum number of pods that can be scheduled above the desired number
     of pods. Value can be an absolute number (ex: 5) or a percentage of desired
     pods (ex: 10%!)(MISSING).
[...]
```

@kubernetes/sig-cli-bugs

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61503

**Special notes for your reviewer**:
/assign @janetkuo @liggitt 
/cc @kubernetes/sig-cli-bugs 

**Release note**:

```release-note
escape literal percent sign when formatting
```

### CODE REVIEW

File: pkg/kubectl/explain/formatter.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ func (f *Formatter) WriteWrapped(str string, a ...interface{}) error {
 	text := fmt.Sprintf(str, a...)
 	strs := wrapString(text, f.Wrap-f.IndentLevel)
 	for _, substr := range strs {
+		// escape literal percent sign
+		substr = strings.Replace(substr, "%", "%%", -1)
```
Comment (soltysh) [inline pkg/kubectl/explain/formatter.go]: Mind adding a test case covering it?

**Review by soltysh — CHANGES_REQUESTED** [review]

File: pkg/kubectl/explain/formatter.go (hunk 1 above)
Comment (liggitt) [reply pkg/kubectl/explain/formatter.go]: Hmm… are there other formatting directives that will cause issues? I'd prefer we stop treating this as a format string altogether than try to escape here for later layers

Comment (apelisse) [reply pkg/kubectl/explain/formatter.go]: Yeah, I think Jordan is right. I'd be better to just take a pre-formatted string, and call `Sprintf` from the caller, or add maybe add a `Writef` for convenience.

**Comment by dixudx** [comment]
@soltysh @liggitt @apelisse Updated. PTAL. Thanks.

File: pkg/kubectl/explain/formatter.go [hunk 2]
```diff
@@ -48,7 +48,11 @@ func (f *Formatter) Write(str string, a ...interface{}) error {
 	for i := 0; i < f.IndentLevel; i++ {
 		indent = indent + " "
 	}
-	_, err := io.WriteString(f.Writer, indent+fmt.Sprintf(str, a...)+"\n")
+
+	if a != nil {
```
Comment (liggitt) [inline pkg/kubectl/explain/formatter.go]: nit, `len(a) > 0`

Comment (dixudx) [reply pkg/kubectl/explain/formatter.go]: @liggitt Done. PTAL. Thanks.

**Pull request merged**